### DISCUSSION TITLE
Zero Line Styling

### DISCUSSION THREAD

**Discussion by wslates-ng** [GSI-CS-CO/chart-fx#382] [Q&A]
Hi,

I have very much enjoyed your library thus far. Great performance and I've been able to achieve just about everything I've wanted to with ease. 

I am using your JDK8 implementation.

However, some of my users have decided that they want to be able to clearly see the zero lines in the chart. In my previous implementation using vanilla JavaFX, I used the `.chart-vertical-zero-line` and the `.chart-horizontal-zero-line` CSS class selectors. I noted that these same class selectors are used in your [chart.css](https://github.com/GSI-CS-CO/chart-fx/blob/master/chartfx-chart/src/main/resources/de/gsi/chart/chart.css) file. However, it doesn't seem to do anything for me when I add styling to the chart. I also couldn't find it used in source code. Is there any way achieve this?

Thanks!

**Comment by RalphSteinhagen** [comment]
@wslates-ng unfortunately these specific CSS definitions seem to be empty stubs and the more pronounced vertical/horizontal lines that go through origin '(0,0)' are not implemented. You can force the axes however that the zero is always included in their auto-computed ranges.

The grid-lines are drawn by the [GidRenderer](https://github.com/GSI-CS-CO/chart-fx/blob/cd202c7c9a0b40f1e66b6d0fe8fd93fffbec5435/chartfx-chart/src/main/java/de/gsi/chart/renderer/spi/GridRenderer.java) that provides the following CSS definitions:

https://github.com/GSI-CS-CO/chart-fx/blob/cd202c7c9a0b40f1e66b6d0fe8fd93fffbec5435/chartfx-chart/src/main/java/de/gsi/chart/renderer/spi/GridRenderer.java#L38-L46

It shouldn't be hard to extend this. Maybe you would be willing to make a PR?

Alternatively, with the existing code-base, you could use either dummy DataSets, the data label renderer, or marker plugins to achieve the same visual effect.


<sub><sup>*N.B. JDK8 became increasingly hard to maintain and increasing incompatibilities w.r.t. JDK11 and beyond (Jigsaw, internal Java API, etc.).   
Please do not see it as an advertisement, but maybe an encouragement to evaluate whether not to move to at least JDK11 since also many other libs that aimed at supporting JDK8 and soon JDK17 are having a hard time bridging that many versions*</sub></sup>

**Reply by wslates-ng** [reply]
Thanks, that's what I thought. Just wanted to make sure I wasn't missing anything. I had considered using a marker, but couldn't figure out how to get it to work on the y-axis. However, I'll look at that closer.

I can't make a PR on company time, but I might be able to do so in my free time soon. :smiley:

_<sup><sup>Don't worry, I will be transitioning to JDK11 sometime soon, probably within the next couple months. :)</sup></sup>_

**Reply by RalphSteinhagen** [reply]
> I can't make a PR on company time, but I might be able to do so in my free time soon. smiley

Looking forward to your PR! :smile: 

May I ask what you are using our lib for? While we wrote this primarily for our own accelerator-control-related use-case, we are open to external contributions and would love to get to know the other use-cases as well... sort of cross-pollination of ideas and experiences that we use to improve. :smile:
